README.md

```diff
@@ -53,6 +53,30 @@ See [lock-filename](#lock-filename) for more information on changing the default
 lockfile paths and [pip-compile-constraint](#pip-compile-constraint) for more
 information on syncing dependency versions across environments.
 
+### Updating Dependencies
+
+Passing arguments like `--upgrade` / `--upgrade-package` to `pip-compile` is
+controlled by two environment variables: `PIP_COMPILE_UPGRADE` and
+`PIP_COMPILE_UPGRADE_PACKAGE`. When either of these environment variables are
+set `hatch` will assume that the environment is out of date and automatically
+run `pip-compile` the next time the environment is activated.
+
+The below command runs `pip-compile --upgrade` on the `default` environment.
+
+```shell
+PIP_COMPILE_UPGRADE=1 hatch env run --env default -- python --version
+```
+
+The below command runs `pip-compile --upgrade-package=mkdocs --upgrade-package=mkdocs-material`
```

Comment:
Also: me, as a huge fan of `pip-compile`, I've never used `--upgrade-package`. In fact I think it's simply a bad thing to ever update only one package at a time? One package could've broken something and another package may have prepared for it by fixing it in a new release, but you're not getting that fix.

So, this mention could be greatly de-emphasized.